### PR TITLE
dialects: (riscv_snitch) Add f32 mul, add, pack from Snitch packed SIMD extension

### PR DESCRIPTION
**Pull request by nazavode**
This PR adds a small set of operations to `riscv_snitch` directly mapping on top of the Snitch custom packed SIMD ISA. Further ops are going to need operand constraints to be correctly register allocated.

### CODE REVIEW

File: tests/filecheck/dialects/riscv_snitch/ops.mlir [hunk 1]
```diff
@@ -86,6 +86,22 @@ riscv_func.func @xdma() {
   riscv_func.return
 }
 
+riscv_func.func @simd() {
+  // CHECK-LABEL: @simd
```
Comment (AntonLydike) [inline tests/filecheck/dialects/riscv_snitch/ops.mlir:90]: Nit: `CHECK-LABEL` is not (yet) supported by our filecheck, we are working on switching to a new one that does support everything though!

**Review by AntonLydike — APPROVED** [review]
Nice work! Just one nitpick about the use of `CHECK-LABEL`

File: tests/filecheck/dialects/riscv_snitch/ops.mlir (hunk 1 above)
Comment (nazavode) [reply tests/filecheck/dialects/riscv_snitch/ops.mlir:90]: Ouch, thanks for spotting!

**Review by superlopuh — COMMENTED** [review]
could you please add assembly printing tests?

**Review by superlopuh — APPROVED** [review]

**Pull request merged**